flash-app.js

```diff
@@ -44,6 +45,8 @@ const STATE_INIT     = 0;
 const STATE_FLASHING = 1;
 const STATE_READING  = 2;
 
+const BOOTLOADER_SIZE_BYTES  = 4096;
```

Comment:
This should be configurable since it may be 8162 bytes too.
See branch [bitrate-detect](https://github.com/crycode-de/mcp-can-boot/tree/bitrate-detect).

Reply:
Thanks. I looked into it. I'd hate to add another required argument for the read command to work right... I think I might just ignore the size of the bootloader and let the full-read terminate early when the MCU returns `CMD_FLASH_READ_ADDRESS_ERROR` [here](https://github.com/hankedan000/mcp-can-boot-flash-app/blob/add-progress-bars/flash-app.js#L477) (this was the previous behavior before I needed to know how far it was going to read). That does mean the progress bar on the full-read will be a little inaccurate since it could end a few pages sooner than expected. I don't think the user will notice the inaccuracy too much. It will probably just look like a quick jump to 100% near the end because it didn't need to read the last few pages of flash.

Perhaps we could eventually extend the bootloader interface to include a command that returns the bootloader's configurable feature set as well as the bootloader size. The size does need to be `#defined` [in the bootloader source](https://github.com/crycode-de/mcp-can-boot/blob/main/src/controllers.h#L19), so it could easily be sent to the flash-app in a response message.

Thoughts? Obviously that would require an update to the bootloader as well, so definitely not going to do that on this PR, but I could capture it in a issue and we'll see where it lands.

Reply:
Okay, first I pushed up a commit that did what I originally said (ignore bootloader size).

Then I got a little creative, and pushed up a 2nd commit with bootloader size querying logic. I'm detecting the bootloader size by extracting the FLASHEND constant that is returned as part of the [CMD_FLASH_ADDRESS_ERROR response](https://github.com/crycode-de/mcp-can-boot/blob/main/src/bootloader.cpp#L282-L292).

Hopefully that's not too much of a hack. Let me know what you think.